src/project/environment.rs

```diff
@@ -137,8 +139,8 @@ impl<'p> Environment<'p> {
     /// If a feature does not specify any channel the default channels from the project metadata are
     /// used instead. However, these are not considered during deduplication. This means the default
     /// channels are always added to the end of the list.
-    pub fn channels(&self) -> IndexSet<&'p Channel> {
```

Comment:
The documentation is not aligned with the code here. currently there is no deduplication, and a name feature will never use the default channels. what should it be?

Reply:
The `true` in `self.features(true)` includes the default feature.

The function returns an `IndexSet` which deduplicates the channels.

Reply:
Yes @baszalmstra.  But default channels are also deduplicated though?

Reply:
See new proposed implementation @baszalmstra